### PR TITLE
feature: automated accessibility testing setup [NP-382]

### PR DESCRIPTION
**Pull request by davidsauntson**
Reasons for change:

Add automated accessibility testing to the design system.

 - installed pa11y and pa11y-ci
 - created /testing/wcag folder with pa11y ci configs
 - created npm script to create pa11y ci configs using backstop scenarios
 - created npm script to run pa11y ci
 - updated design system:
    - header lang link to remove WCAG failure
    - version history & number
 - added tasks into package.json

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
-## <sub>v1.9.0</sub>
+## <sub>v1.8.3</sub>
```
Comment (panosvoudouris) [inline CHANGELOG.md:1]: Something fishy happening here?

File: scripts/wcag-test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// use correct pa11y config based on args provided
+const ci = process.argv.some((arg) => arg === '--ci');
+const baseCommand = 'cd wcag && npx pa11y-ci';
+const command = ci
+  ? `${baseCommand} --config .pa11yci.ci.json`
+  : `${baseCommand} --config .pa11yci.local.json`;
+
+// update pa11y configs using data in vr backstop scenarios
+ci ? createCiPa11yConfigs() : createLocalPa11yConfigs();
```
Comment (baseonmars) [inline scripts/wcag-test.js:21]: not sure how this passed lint - use ternery statements when you want to return a value. Just use a boring if statement here.

**Review by panosvoudouris — CHANGES_REQUESTED** [review]

**Review by baseonmars — COMMENTED** [review]
I'm gonna check it out and confirm it works.

File: package.json [hunk 3]
```diff
@@ -1,127 +1,129 @@
...
-        "*.scss": [
-            "stylelint"
-        ]
-    },
-    "husky": {
-        "hooks": {
-            "pre-commit": "npm run pre-commit"
-        }
+  "name": "@citizensadvice/design-system",
+  "version": "1.8.3",
```
Comment (baseonmars) [inline package.json:3]: This version needs to be 1.9.0 too.

**Review by baseonmars — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -28,10 +28,20 @@ The `check-size` script will check the current output and compare it to the last
...
 As this is just a CSS library to test it we use storybook. The concept is that a test version of storybook is built which is then used to do screenshot comparison in BackstopJS. BackstopJS automates visual regression testing of your responsive web UI by comparing DOM screenshots over time.
 
 You can install backstop by running `npm run vr-test:install`, this will install the visual regression testing tools. These are installed under `/testing` and have a separate package to the main Design System, this is done to keep the install time of the Design System low.
 
+#### Accessibility
+
+`pa11y-ci` is used to perform automated accessibility tests.  The default configuration is to use htmlsniffer and the urls in the `.pa11yci.*.json` config files.  The files are generated dynamically from the backtopJS scenarios when running the `wcag` tasks in `package.json`.
+
+The following tasks can be used to perform WCAG tests:
+  - `npm run wcag-test` &rarr; runs WCAG tests using the local config files.  This runs storybook then runs `pa11y-ci` against the local storybook urls
```
Comment (luke-hill) [inline CONTRIBUTING.md:42]: Not sure if this is encoding or whatnot, but this says `&rarr;` should it say that?

File: testing/wcag/.pa11yci.ci.json [hunk 5]
```diff
@@ -0,0 +1 @@
...cker.internal:6006/iframe.html?id=3-components--input","http://host.docker.internal:6006/iframe.html?id=3-components--logo","http://host.docker.internal:6006/iframe.html?id=3-components--navigation","http://host.docker.internal:6006/iframe.html?id=3-components--notice-banner","http://host.docker.internal:6006/iframe.html?id=3-components--page-review","http://host.docker.internal:6006/iframe.html?id=3-components--pagination","http://host.docker.internal:6006/iframe.html?id=3-components--paging-info","http://host.docker.internal:6006/iframe.html?id=3-components--radio-group","http://host.docker.internal:6006/iframe.html?id=3-components--radio-group-small","http://host.docker.internal:6006/iframe.html?id=3-components--search","http://host.docker.internal:6006/iframe.html?id=3-components--success-message","http://host.docker.internal:6006/iframe.html?id=3-components--targeted-content","http://host.docker.internal:6006/iframe.html?id=3-components--textarea","http://host.docker.internal:6006/iframe.html?id=3-components--website-feedback","http://host.docker.internal:6006/iframe.html?id=3-components--related-content","http://host.docker.internal:6006/iframe.html?id=3-components--tables"]}
```
Comment (luke-hill) [inline testing/wcag/.pa11yci.ci.json:1]: can we make these multi-lines? Or do they need to be one line?

**Review by luke-hill — COMMENTED** [review]
I've only ran pa11y using CLI args. But it was a lot thinner than this. So maybe I'm wrong but

How do we specify runners?
How do we specify report format?

File: package.json (hunk 3 above)
Comment (panosvoudouris) [reply package.json]: In general, the package version should not be changed manually. The `release` script will sort all that and update the CHANGELOG.

File: testing/wcag/.pa11yci.ci.json (hunk 5 above)
Comment (davidsauntson) [reply testing/wcag/.pa11yci.ci.json:1]: it's automatically generated by the task, so it doesn't really matter.

**Comment by davidsauntson** [comment]
> I've only ran pa11y using CLI args. But it was a lot thinner than this. So maybe I'm wrong but
> 
> How do we specify runners?
> How do we specify report format?

Runners can be added in the `baseConfig` object in the `update-pa11y-config.js` script - at the moment what I've done doesn't support changing the runner or reporting format via args as there isn't really a need.... or is there? If so we should document in a ticket and pick it up separately.

**Review by baseonmars — APPROVED** [review]
LGTM

**Review by panosvoudouris — APPROVED** [review]
👍

**Pull request merged**